cdap-formats/src/main/java/io/cdap/cdap/format/StructuredRecordStringConverter.java

```diff
@@ -174,7 +175,11 @@ protected static void handleFieldConversion(StructuredRecord.Builder builder,
     // Decode base64 payload for byte fields.
     if (fieldSchema.getType() == Schema.Type.BYTES) {
       try {
-        builder.set(fieldName, FormatUtils.base64Decode(part));
+        if (fieldSchema.getLogicalType() == Schema.LogicalType.FIXED) {
+          builder.set(fieldName, ByteBuffer.wrap((part).getBytes()));
```

Comment:
You should still base64 decode the input. The byte array is represented as a string, so you need to base64 decode it to get a byte array, and then you can check the length.

Reply:
Done